core/src/calleeFunctions/helpers/oneInch.ts

```diff
@@ -0,0 +1,161 @@
+import { CollateralConfig } from '../../types';
+import BigNumber from '../../bignumber';
+import { getTokenAddressByNetworkAndSymbol } from '../../tokens';
+import { WAD_NUMBER_OF_DIGITS } from '../../constants/UNITS';
+import { RateLimiter } from 'limiter';
+
+const REQUEST_LIMITER = new RateLimiter({ tokensPerInterval: 1, interval: 'second' });
+const EXPECTED_SIGNATURE = '0x12aa3caf'; // see https://www.4byte.directory/signatures/?bytes4_signature=0x12aa3caf
+const SUPPORTED_1INCH_NETWORK_IDS = [1, 56, 137, 10, 42161, 100, 43114];
+
+export const getOneInchUrl = (chainId: number) => {
+    return `https://api.1inch.io/v5.0/${chainId}`;
+};
+
+interface Protocol {
+    id: string;
+    title: string;
+    img: string;
+    img_color: string;
+}
+
+interface OneInchToken {
+    symbol: string;
+    name: string;
+    address: string;
+    decimals: 0;
+    logoURI: string;
+}
+interface OneInchSwapRepsonse {
+    fromToken: OneInchToken;
+    toToken: OneInchToken;
+    toTokenAmount: string;
+    fromTokenAmount: string;
+    protocols: OneInchSwapRoute[];
+    tx: {
+        from: string;
+        to: string;
+        data: string;
+        value: string;
+        gasPrice: string;
+        gas: string;
+    };
+}
+interface LiquiditySourcesResponse {
+    protocols: Protocol[];
+}
+type OneInchSwapRoute = { name: string; part: number; fromTokenAddress: string; toTokenAddress: string }[][];
+
+export const executeOneInchApiRequest = async (
+    chainId: number,
+    endpoint: '/swap' | '/quote' | '/liquidity-sources',
+    params?: Record<string, any>
+) => {
+    const oneInchUrl = getOneInchUrl(chainId);
+    const url = `${oneInchUrl}${endpoint}?${new URLSearchParams(params)}`;
+    await REQUEST_LIMITER.removeTokens(1);
+    const response = await fetch(url).then(res => res.json());
+    if (response.error) {
+        throw new Error(`failed to receive response from oneinch: ${response.error}`);
+    }
+    return response;
+};
+
+export async function getOneinchValidProtocols(chainId: number) {
+    // Fetch all supported protocols except for the limit orders
+    const response: LiquiditySourcesResponse = await executeOneInchApiRequest(chainId, '/liquidity-sources');
+    const protocolIds = response.protocols.map(protocol => protocol.id);
+    return protocolIds.filter(protocolId => !protocolId.toLowerCase().includes('limit'));
+}
+
+export async function getOneinchSwapParameters(
+    network: string,
+    collateralSymbol: string,
+    amount: string,
+    marketId: string,
+    slippage = '10'
+): Promise<OneInchSwapRepsonse> {
+    const isFork = getNetworkConfigByType(network).isFork;
+    const chainId = isFork ? 1 : parseInt(getChainIdByNetworkType(network) || '', 16);
+    if (!isFork && !SUPPORTED_1INCH_NETWORK_IDS.includes(chainId)) {
+        throw new Error(`1inch does not support network ${network}`);
+    }
+    if (Number.isNaN(chainId)) {
```

Comment:
When can this happen? Can you then please move this error to the `networks.ts` function

Reply:
removed due to prev comment adjustment